v0.18.0

DJ2LS opened the pull request.

DJ2LS committed:
attempt of streaming audio to gui

DJ2LS committed:
attempt of streaming audio to gui

DJ2LS committed:
adjusted streaming

DJ2LS committed:
adjusted streaming

DJ2LS committed:
buffer adjustments

DJ2LS committed:
buffer adjustments

DJ2LS committed:
adjusted streaming

DJ2LS committed:
adjusted streaming

DJ2LS committed:
adjusted streaming

DJ2LS committed:
adjusted streaming

DJ2LS committed:
adjusted streaming

DJ2LS committed:
first basic work on NORM implementation

DJ2LS committed:
save data to database

DJ2LS committed:
save data to database

DJ2LS committed:
work on multi burst

DJ2LS committed:
fixing crc failure

DJ2LS committed:
update status if crc failure or received

DJ2LS committed:
Merge branch 'develop' into ls-norm

DJ2LS committed:
first parts of implementing gui stuff

DJ2LS committed:
adjust database

DJ2LS committed:
get broadcasts per domain

DJ2LS committed:
adjusted gui

DJ2LS committed:
adjusted database

DJ2LS committed:
first working transmission via gui

DJ2LS committed:
fixing smaller stuff

DJ2LS committed:
fixing smaller stuff

DJ2LS committed:
updated database

DJ2LS committed:
adjusted new broadcast

DJ2LS committed:
adjusted new broadcast

DJ2LS committed:
enable datac4 listening and fix database inserting on IRS

DJ2LS committed:
adjusted payloads

DJ2LS committed:
first work on NACK

DJ2LS committed:
first work on NACK

Commit:
work on nack

DJ2LS committed:
Merge branch 'develop' into ls-audio

DJ2LS committed:
Merge branch 'develop' into ls-norm

DJ2LS committed:
Merge branch 'develop' into ls-norm

DJ2LS committed:
Merge branch 'develop' into ls-audio

DJ2LS committed:
Merge branch 'develop' into ls-norm

DJ2LS committed:
fix iss transmission

Commit:
work on nack

DJ2LS committed:
fix iss transmission

DJ2LS committed:
adjust database - added attempts

DJ2LS committed:
check for max attempts

DJ2LS committed:
run nack only if next transmission is reached

DJ2LS committed:
work on timezone stuff... :-/

DJ2LS committed:
work on timezone stuff... :-/

DJ2LS committed:
fix transmission stuff

DJ2LS committed:
change timestamps to unix timestamps

DJ2LS committed:
change timestamps to unix timestamps

DJ2LS committed:
fixing message type

DJ2LS committed:
fixing info modal

DJ2LS committed:
add message placeholder when receiving

DJ2LS committed:
limit input data

DJ2LS committed:
remove support for python3.8

DJ2LS committed:
time related fixed and logging

DJ2LS committed:
bump version

DJ2LS committed:
adjusted backoff intervall

DJ2LS committed:
adjusted backoff intervall

DJ2LS committed:
adjusted backoff intervall

DJ2LS committed:
adjusted backoff intervall

DJ2LS committed:
fix sending of repair bursts

DJ2LS committed:
Merge branch 'develop' into ls-norm

DJ2LS committed:
Merge branch 'develop' into ls-norm

DJ2LS committed:
Merge branch 'develop' into ls-audio

dk5sm committed:
Started translation for new broadcast feature

dk5sm committed:
Added translation for new broadcast feature

dk5sm committed:
Fix for new broadcast translation

DJ2LS committed:
fixing translation string

dk5sm committed:
small translation fix

added modal.startnewbroadcast

dk5sm committed:
added "deletebroadcastdomain" translation

DJ2LS committed:
work on retransmitting data

DJ2LS committed:
Merge remote-tracking branch 'origin/ls-norm' into ls-norm

dk5sm committed:
added new translations for broadcast feature

dk5sm committed:
minor translation corrections

DJ2LS committed:
small bug fixes

DJ2LS committed:
fixing timestamps

DJ2LS committed:
run retransmission async

DJ2LS committed:
run retransmission async

DJ2LS committed:
Merge branch 'develop' into ls-norm

DJ2LS committed:
Merge branch 'develop' into ls-audio

DJ2LS committed:
unread counter...

DJ2LS committed:
respect message expired

DJ2LS committed:
Merge branch 'develop' into ls-norm

DJ2LS committed:
Merge branch 'develop' into ls-audio

DJ2LS committed:
timestamp related fix

DJ2LS committed:
Merge remote-tracking branch 'origin/ls-norm' into ls-norm

dk5sm committed:
Inserted latest broadcast-related translations

dk5sm committed:
Update and corrections

DJ2LS committed:
Merge branch 'develop' into ls-norm

DJ2LS committed:
adjust domain list

DJ2LS committed:
Merge branch 'develop' into ls-norm

DJ2LS committed:
Merge remote-tracking branch 'origin/ls-norm' into ls-norm

DJ2LS committed:
adjust validation

DJ2LS committed:
Merge branch 'develop' into ls-audio

DJ2LS committed:
Merge branch 'develop' into ls-norm

DJ2LS committed:
Merge branch 'develop' into ls-audio

DJ2LS committed:
adjusting repair

DJ2LS committed:
implement METAR message for showing different message types

DJ2LS committed:
ICAO override

DJ2LS committed:
Merge branch 'develop' into ls-norm

DJ2LS committed:
Merge branch 'develop' into ls-audio

DJ2LS committed:
Merge pull request #1012 from DJ2LS/develop

Develop

DJ2LS committed:
update workflow

dependabot[bot] committed:
Bump actions/setup-node from 4 to 5

Bumps [actions/setup-node](https://github.com/actions/setup-node) from 4 to 5.
- [Release notes](https://github.com/actions/setup-node/releases)
- [Commits](https://github.com/actions/setup-node/compare/v4...v5)

---
updated-dependencies:
- dependency-name: actions/setup-node
  dependency-version: '5'
  dependency-type: direct:production
  update-type: version-update:semver-major
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
Bump actions/setup-python from 5 to 6

Bumps [actions/setup-python](https://github.com/actions/setup-python) from 5 to 6.
- [Release notes](https://github.com/actions/setup-python/releases)
- [Commits](https://github.com/actions/setup-python/compare/v5...v6)

---
updated-dependencies:
- dependency-name: actions/setup-python
  dependency-version: '6'
  dependency-type: direct:production
  update-type: version-update:semver-major
...

Signed-off-by: dependabot[bot] <support@github.com>

as3ii committed:
Added pyproject.toml

as3ii committed:
Removed useless/deprecated dependencies, updated nuitka

as3ii committed:
Deprecate python versions older then 3.10, add 3.13 where missing

DJ2LS committed:
Merge pull request #1018 from as3ii/fix-py-side

Add pyproject, cleanup python dependencies, deprecate python 3.9

DJ2LS committed:
Merge branch 'develop' into dependabot/github_actions/develop/actions/setup-node-5

DJ2LS committed:
Merge branch 'develop' into dependabot/github_actions/develop/actions/setup-python-6

DJ2LS committed:
Merge branch 'develop' into ls-audio

DJ2LS committed:
Merge branch 'develop' into ls-norm

DJ2LS committed:
Merge pull request #1014 from DJ2LS/dependabot/github_actions/develop/actions/setup-node-5

Bump actions/setup-node from 4 to 5

DJ2LS committed:
Merge branch 'develop' into dependabot/github_actions/develop/actions/setup-python-6

erikdewildt committed:
Fix English locale hamlib radio control help texts and add Dutch
translation

erikdewildt committed:
Rename dutch local file to nl_Nederlands

as3ii committed:
Fixed typo in pyproject.toml

as3ii committed:
Install only required dependencies

- `pip install .` installs only the libraries required to run freedata
- `pip install .[test]` installs the libraries required for running and testing freedata
- `pip install .[dev]` installs the libraries required for running, testing and linting/checking freedata
- `pip install .[build]` installs only the libraries required to run and build the freedata pip module
- `pip install .[nuitka]` installs only the libraries required to run and build the freedata executable

requirements.txt remains only for backward compatibility

erikdewildt committed:
Merge branch 'develop' into add_dutch_language

erikdewildt committed:
Add distance unit setting to choose between Kilometer or Miles.

erikdewildt committed:
Update locales for distance unit setting.

erikdewildt committed:
Merge branch 'DJ2LS:main' into distance_unit_setting

DJ2LS committed:
Merge pull request #1015 from DJ2LS/dependabot/github_actions/develop/actions/setup-python-6

Bump actions/setup-python from 5 to 6

DJ2LS committed:
Merge pull request #1021 from erikdewildt/add_dutch_language

Fix English locale hamlib radio control help texts and add Dutch language

DJ2LS committed:
Merge branch 'main' into develop

erikdewildt committed:
Add a support for Nautical Miles

erikdewildt committed:
Merge remote-tracking branch 'origin/develop' into distance_unit_setting

erikdewildt committed:
Update Dutch locale for distance unit settting.

erikdewildt committed:
Merge branch 'distance_unit_setting' of github.com:erikdewildt/FreeDATA into distance_unit_setting

dependabot[bot] committed:
Bump actions/setup-node from 5 to 6

Bumps [actions/setup-node](https://github.com/actions/setup-node) from 5 to 6.
- [Release notes](https://github.com/actions/setup-node/releases)
- [Commits](https://github.com/actions/setup-node/compare/v5...v6)

---
updated-dependencies:
- dependency-name: actions/setup-node
  dependency-version: '6'
  dependency-type: direct:production
  update-type: version-update:semver-major
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
Bump github/codeql-action from 3 to 4

Bumps [github/codeql-action](https://github.com/github/codeql-action) from 3 to 4.
- [Release notes](https://github.com/github/codeql-action/releases)
- [Changelog](https://github.com/github/codeql-action/blob/main/CHANGELOG.md)
- [Commits](https://github.com/github/codeql-action/compare/v3...v4)

---
updated-dependencies:
- dependency-name: github/codeql-action
  dependency-version: '4'
  dependency-type: direct:production
  update-type: version-update:semver-major
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
Bump actions/upload-artifact from 4 to 5

Bumps [actions/upload-artifact](https://github.com/actions/upload-artifact) from 4 to 5.
- [Release notes](https://github.com/actions/upload-artifact/releases)
- [Commits](https://github.com/actions/upload-artifact/compare/v4...v5)

---
updated-dependencies:
- dependency-name: actions/upload-artifact
  dependency-version: '5'
  dependency-type: direct:production
  update-type: version-update:semver-major
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
Update nuitka requirement from <=2.7.16 to <=2.8.4

Updates the requirements on [nuitka](https://github.com/Nuitka/Nuitka) to permit the latest version.
- [Changelog](https://github.com/Nuitka/Nuitka/blob/develop/Changelog.rst)
- [Commits](https://github.com/Nuitka/Nuitka/compare/0.3.11a...2.8.4)

---
updated-dependencies:
- dependency-name: nuitka
  dependency-version: 2.8.4
  dependency-type: direct:production
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
Bump @babel/eslint-parser from 7.28.4 to 7.28.5 in /freedata_gui

Bumps [@babel/eslint-parser](https://github.com/babel/babel/tree/HEAD/eslint/babel-eslint-parser) from 7.28.4 to 7.28.5.
- [Release notes](https://github.com/babel/babel/releases)
- [Changelog](https://github.com/babel/babel/blob/main/CHANGELOG.md)
- [Commits](https://github.com/babel/babel/commits/v7.28.5/eslint/babel-eslint-parser)

---
updated-dependencies:
- dependency-name: "@babel/eslint-parser"
  dependency-version: 7.28.5
  dependency-type: direct:development
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
Bump core-js from 3.45.1 to 3.46.0 in /freedata_gui

Bumps [core-js](https://github.com/zloirock/core-js/tree/HEAD/packages/core-js) from 3.45.1 to 3.46.0.
- [Release notes](https://github.com/zloirock/core-js/releases)
- [Changelog](https://github.com/zloirock/core-js/blob/master/CHANGELOG.md)
- [Commits](https://github.com/zloirock/core-js/commits/v3.46.0/packages/core-js)

---
updated-dependencies:
- dependency-name: core-js
  dependency-version: 3.46.0
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

DJ2LS committed:
Merge pull request #1025 from DJ2LS/dependabot/github_actions/develop/actions/setup-node-6

DJ2LS committed:
Merge pull request #1026 from DJ2LS/dependabot/github_actions/develop/github/codeql-action-4

DJ2LS committed:
Merge pull request #1027 from DJ2LS/dependabot/github_actions/develop/actions/upload-artifact-5

DJ2LS committed:
Merge pull request #1028 from DJ2LS/dependabot/pip/develop/nuitka-lte-2.8.4

DJ2LS committed:
Merge pull request #1030 from DJ2LS/dependabot/npm_and_yarn/freedata_gui/develop/babel/eslint-parser-7.28.5

DJ2LS committed:
Merge pull request #1031 from DJ2LS/dependabot/npm_and_yarn/freedata_gui/develop/core-js-3.46.0

dependabot[bot] committed:
Bump @babel/core from 7.28.4 to 7.28.5 in /freedata_gui

Bumps [@babel/core](https://github.com/babel/babel/tree/HEAD/packages/babel-core) from 7.28.4 to 7.28.5.
- [Release notes](https://github.com/babel/babel/releases)
- [Changelog](https://github.com/babel/babel/blob/main/CHANGELOG.md)
- [Commits](https://github.com/babel/babel/commits/v7.28.5/packages/babel-core)

---
updated-dependencies:
- dependency-name: "@babel/core"
  dependency-version: 7.28.5
  dependency-type: direct:development
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
Bump eslint from 9.37.0 to 9.39.0 in /freedata_gui

Bumps [eslint](https://github.com/eslint/eslint) from 9.37.0 to 9.39.0.
- [Release notes](https://github.com/eslint/eslint/releases)
- [Commits](https://github.com/eslint/eslint/compare/v9.37.0...v9.39.0)

---
updated-dependencies:
- dependency-name: eslint
  dependency-version: 9.39.0
  dependency-type: direct:development
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

DJ2LS committed:
Merge pull request #1032 from DJ2LS/dependabot/npm_and_yarn/freedata_gui/develop/babel/core-7.28.5

DJ2LS committed:
Merge pull request #1033 from DJ2LS/dependabot/npm_and_yarn/freedata_gui/develop/eslint-9.39.0

DJ2LS committed:
Merge pull request #1023 from erikdewildt/distance_unit_setting

as3ii committed:
Removed setup.py, fixed build of pip module

as3ii committed:
Reordered python dependencies, removed useless pyinstaller

DJ2LS committed:
Merge pull request #1022 from as3ii/project-update

as3ii committed:
Switched to ruff for linting/formatting, added ruff CI

DJ2LS committed:
Merge branch 'develop' into ls-audio

DJ2LS committed:
Merge branch 'develop' into ls-norm

as3ii committed:
Added basic CONTRIBUTING.md

as3ii committed:
Run `ruff format`

This applies mostly the same rules that `black` used.

This commit should not have introduced any logical or functional
changes, only fixing style consistency

as3ii committed:
Removed references to pylint and useless file encoding

as3ii committed:
Switched from editing `sys.path` to proper PEP8 imports

See: https://peps.python.org/pep-0008/#imports

Here absolute imports have been used to avoid the error
`ImportError: attempted relative import with no known parent package`.

as3ii committed:
Fixed ruff rule E401 and E402

See https://peps.python.org/pep-0008/#imports

as3ii committed:
Fixed ruff rule E713, E721 and E722

See https://peps.python.org/pep-0008/#programming-recommendations

as3ii committed:
Removed unused imports

Ruff's rule F401

as3ii committed:
Fixed ruff F403 and F405

Now all F4 rules are fixed

as3ii committed:
Removed f-strings without placeholders (ruff F541)

as3ii committed:
Fixed errors in create_custom_ofdm_mod

as3ii committed:
Fixed `undefined name` errors (ruff F812)

as3ii committed:
Substituted unused variables with wildcard (ruff F841)

See https://peps.python.org/pep-0634/#wildcard-pattern

as3ii committed:
Removed redefined set_tuner() flrig's function (ruff F811)

as3ii committed:
Executed `ruff format --preview`, small update to CONTRIBUTING.md

dependabot[bot] committed:
Bump globals from 16.4.0 to 16.5.0 in /freedata_gui

Bumps [globals](https://github.com/sindresorhus/globals) from 16.4.0 to 16.5.0.
- [Release notes](https://github.com/sindresorhus/globals/releases)
- [Commits](https://github.com/sindresorhus/globals/compare/v16.4.0...v16.5.0)

---
updated-dependencies:
- dependency-name: globals
  dependency-version: 16.5.0
  dependency-type: direct:development
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
Bump eslint from 9.39.0 to 9.39.1 in /freedata_gui

Bumps [eslint](https://github.com/eslint/eslint) from 9.39.0 to 9.39.1.
- [Release notes](https://github.com/eslint/eslint/releases)
- [Commits](https://github.com/eslint/eslint/compare/v9.39.0...v9.39.1)

---
updated-dependencies:
- dependency-name: eslint
  dependency-version: 9.39.1
  dependency-type: direct:development
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
Bump bootstrap-vue-next from 0.40.5 to 0.40.9 in /freedata_gui

Bumps [bootstrap-vue-next](https://github.com/bootstrap-vue-next/bootstrap-vue-next/tree/HEAD/packages/bootstrap-vue-next) from 0.40.5 to 0.40.9.
- [Release notes](https://github.com/bootstrap-vue-next/bootstrap-vue-next/releases)
- [Changelog](https://github.com/bootstrap-vue-next/bootstrap-vue-next/blob/main/packages/bootstrap-vue-next/CHANGELOG.md)
- [Commits](https://github.com/bootstrap-vue-next/bootstrap-vue-next/commits/nuxt-v0.40.9/packages/bootstrap-vue-next)

---
updated-dependencies:
- dependency-name: bootstrap-vue-next
  dependency-version: 0.40.9
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
Bump marked from 16.3.0 to 17.0.1 in /freedata_gui

Bumps [marked](https://github.com/markedjs/marked) from 16.3.0 to 17.0.1.
- [Release notes](https://github.com/markedjs/marked/releases)
- [Commits](https://github.com/markedjs/marked/compare/v16.3.0...v17.0.1)

---
updated-dependencies:
- dependency-name: marked
  dependency-version: 17.0.1
  dependency-type: direct:production
  update-type: version-update:semver-major
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
Bump actions/checkout from 5 to 6

Bumps [actions/checkout](https://github.com/actions/checkout) from 5 to 6.
- [Release notes](https://github.com/actions/checkout/releases)
- [Changelog](https://github.com/actions/checkout/blob/main/CHANGELOG.md)
- [Commits](https://github.com/actions/checkout/compare/v5...v6)

---
updated-dependencies:
- dependency-name: actions/checkout
  dependency-version: '6'
  dependency-type: direct:production
  update-type: version-update:semver-major
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
Update nuitka requirement from <=2.8.4 to <=2.8.9

Updates the requirements on [nuitka](https://github.com/Nuitka/Nuitka) to permit the latest version.
- [Changelog](https://github.com/Nuitka/Nuitka/blob/develop/Changelog.rst)
- [Commits](https://github.com/Nuitka/Nuitka/compare/0.3.11a...2.8.9)

---
updated-dependencies:
- dependency-name: nuitka
  dependency-version: 2.8.9
  dependency-type: direct:production
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
Bump actions/upload-artifact from 5 to 6

Bumps [actions/upload-artifact](https://github.com/actions/upload-artifact) from 5 to 6.
- [Release notes](https://github.com/actions/upload-artifact/releases)
- [Commits](https://github.com/actions/upload-artifact/compare/v5...v6)

---
updated-dependencies:
- dependency-name: actions/upload-artifact
  dependency-version: '6'
  dependency-type: direct:production
  update-type: version-update:semver-major
...

Signed-off-by: dependabot[bot] <support@github.com>

DJ2LS committed:
Merge pull request #1034 from as3ii/ruff

Switched to ruff for linting/formatting, added ruff CI, added CONTIRBUTING.md

DJ2LS committed:
Merge pull request #1039 from DJ2LS/dependabot/npm_and_yarn/freedata_gui/develop/globals-16.5.0

Bump globals from 16.4.0 to 16.5.0 in /freedata_gui

DJ2LS committed:
Merge pull request #1040 from DJ2LS/dependabot/npm_and_yarn/freedata_gui/develop/eslint-9.39.1

Bump eslint from 9.39.0 to 9.39.1 in /freedata_gui

DJ2LS committed:
Merge pull request #1042 from DJ2LS/dependabot/npm_and_yarn/freedata_gui/develop/bootstrap-vue-next-0.40.9

Bump bootstrap-vue-next from 0.40.5 to 0.40.9 in /freedata_gui

DJ2LS committed:
Merge pull request #1043 from DJ2LS/dependabot/npm_and_yarn/freedata_gui/develop/marked-17.0.1

Bump marked from 16.3.0 to 17.0.1 in /freedata_gui

DJ2LS committed:
Merge pull request #1044 from DJ2LS/dependabot/github_actions/develop/actions/checkout-6

Bump actions/checkout from 5 to 6

DJ2LS committed:
Merge pull request #1045 from DJ2LS/dependabot/pip/develop/nuitka-lte-2.8.9

Update nuitka requirement from <=2.8.4 to <=2.8.9

DJ2LS committed:
Merge pull request #1047 from DJ2LS/dependabot/github_actions/develop/actions/upload-artifact-6

Bump actions/upload-artifact from 5 to 6

DJ2LS committed:
Merge branch 'develop' into ls-norm

Commit:
make norm a setting - trying to merge this PR soon

Commit:
fixing some path related issues - trying to merge this PR soon

DJ2LS committed:
Merge pull request #989 from DJ2LS/ls-norm

R&D NORM Groupchat

DJ2LS committed:
Merge branch 'develop' into ls-audio

DJ2LS committed:
Merge pull request #987 from DJ2LS/ls-audio

R&D audio streaming

Commit:
bump version

Commit:
removed stats publishing

Commit:
bump version in package json

DJ2LS committed:
Merge branch 'main' into develop

DJ2LS committed:
Delete freedata_server/nmea.py

DJ2LS commented:
TODO: Check if windows build is still working and fixing failing tests ....

as3ii committed:
Increased max line length to 120, reformatted everything

In practice, in this commit I've increased from 100 to 120 the maximum
allowed line length for python files and re-run ruff's formatter.

as3ii committed:
Fixed ruff E7

as3ii committed:
Fixed ruff F541

as3ii committed:
Fixed ruff F841

Commit:
support for M1+ chips

OH2JRB commented:
> root@sandbox:~/freedata# bash ./install-freedata-linux.sh develop 4.6.5
> root@sandbox:~/freedata# bash ./run-freedata-linux.sh

*************************************************************************
Running the FreeDATA server processes
*************************************************************************
Traceback (most recent call last):
  File "/root/freedata/FreeDATA/freedata_server/server.py", line 12, in <module>
    from freedata_server.log_handler import setup_logging
ModuleNotFoundError: No module named 'freedata_server'

root@sandbox:~/freedata# python3 --version
Python 3.13.5

Is this Python-version-specific?

Commit:
fixed another wrong path

DJ2LS commented:
> > root@sandbox:~/freedata# bash ./install-freedata-linux.sh develop 4.6.5
> > root@sandbox:~/freedata# bash ./run-freedata-linux.sh
> 
> Running the FreeDATA server processes
> 
> Traceback (most recent call last): File "/root/freedata/FreeDATA/freedata_server/server.py", line 12, in from freedata_server.log_handler import setup_logging ModuleNotFoundError: No module named 'freedata_server'
> 
> root@sandbox:~/freedata# python3 --version Python 3.13.5
> 
> Is this Python-version-specific?

It's probably because of the pytoml file and the related changes to some file imports. 
Are you running latest bash file? What happens if you are running server.py manually?

DJ2LS commented:
Can't reproduce the issue above on Ubuntu and MacOS. @as3ii @OH2JRB

Mashintime commented:
> No module named 'freedata_server'

You need to install the python requirements in the correct/new way.  i.e. while in the root dir, `python -m pip install .`

See for more info:  https://github.com/DJ2LS/FreeDATA/blob/develop-ruff/CONTRIBUTING.md

OH2JRB commented:
So the information on https://wiki.freedata.app/en/installation/linux-install-script is obsoleted until the script is reworked.

as3ii commented:
probably the installer must be updated as Mashintime suggested and, I guess, we can take advantage of this to remove the installation of colorama and pyaudio via system package manager as those packages are installed via pip.

@OH2JRB the instructions on the wiki are valid for the latest release, not for the development branch

OH2JRB commented:
@as3ii I partly disagree - the install script description on the web page says "There are 3 installation options available: "default", "development", "specific""

bash install-freedata-linux.sh (FreeDATA-branch) (hamlib-version)

DJ2LS commented:
@deejdeej could you do us the favour and updating the installation scripts and also adding Ubuntu 24.10 and 25.04? Thanks 🙂

as3ii commented:
Oh, I've noticed just now that in the new CONTRIBUTING.md are missing instructions regarding codec2 (not a serious issue because freedata have in its sources the compiled library) and hamlib (for recent distros the distro-provided one probably is good enough: 4.5 on debian 12 and ubuntu 24.05, 4.6 on debian 13 and ubuntu 25.04, etc.).
In the next few days I can update it, any preferences or criticisms?

In the meantime I'm working on the ruff fixes

as3ii committed:
Fixed ruff F811 and F401

deejdeej committed:
added Ubuntu 24.10 and 25.04 - changed default hamlib version to 4.6.5

deejdeej commented:
> @deejdeej could you do us the favour and updating the installation scripts and also adding Ubuntu 24.10 and 25.04? Thanks 🙂

I added Ubuntu 24.10 and 25.04 to the Linux script in dev, and changed the default hamlib version in the script from 4.5.5 to 4.6.5.

Please note I don't have Ubuntu 24.10 or 25.04 here so would appreciate someone else testing to make sure these work properly.

DJ2LS commented:
Thanks 🙏 Tested 25.10 today - working fine. Biggest issue is actually the adjustment of the installation scripts for supporting the pyproject file so we can merge this big PR soon.

deejdeej committed:
add pyproject.toml support

deejdeej commented:
Ok, I think what you are asking is to change:
pip install --upgrade -r requirements.txt
to
python -m pip install .

in the Linux install script, which I've just done.

Please note this may break the instructions for installing the "main" branch that we have listed at
https://wiki.freedata.app/en/installation/linux-install-script

On that page change these two lines:
wget -N https://raw.githubusercontent.com/DJ2LS/FreeDATA/develop/tools/Linux/install-freedata-linux.sh
wget -N https://raw.githubusercontent.com/DJ2LS/FreeDATA/develop/tools/Linux/run-freedata-linux.sh

to

wget -N https://raw.githubusercontent.com/DJ2LS/FreeDATA/main/tools/Linux/install-freedata-linux.sh
wget -N https://raw.githubusercontent.com/DJ2LS/FreeDATA/main/tools/Linux/run-freedata-linux.sh

as3ii committed:
Fixed other module's paths

DJ2LS commented:
Ah, I see, thanks for the reminder @deejdeej ! This should have been changed earlier. I will try doing this tomorrow. Thanks for the fast update of the scripts!

DJ2LS committed:
Merge pull request #1052 from as3ii/ruff2

DJ2LS commented:
> Ok, I think what you are asking is to change: pip install --upgrade -r requirements.txt to python -m pip install .
> 
> in the Linux install script, which I've just done.
> 
> Please note this may break the instructions for installing the "main" branch that we have listed at https://wiki.freedata.app/en/installation/linux-install-script
> 
> On that page change these two lines: wget -N https://raw.githubusercontent.com/DJ2LS/FreeDATA/develop/tools/Linux/install-freedata-linux.sh wget -N https://raw.githubusercontent.com/DJ2LS/FreeDATA/develop/tools/Linux/run-freedata-linux.sh
> 
> to
> 
> wget -N https://raw.githubusercontent.com/DJ2LS/FreeDATA/main/tools/Linux/install-freedata-linux.sh wget -N https://raw.githubusercontent.com/DJ2LS/FreeDATA/main/tools/Linux/run-freedata-linux.sh

Done

DJ2LS committed:
Merge branch 'main' into develop

DJ2LS commented:
It seems, we are having some path related issues: 

The message-database won't be loaded until explicitly setting the path - so we have to check the mechanism for the default database path. 

Sending a new Broadcasting message isn't working - freedata_server is not found. So it's also a path related issue, not occurring when running in IDE, but from scripts or manually, outside an IDE

dependabot[bot] committed:
Bump actions/checkout from 4 to 6

Bumps [actions/checkout](https://github.com/actions/checkout) from 4 to 6.
- [Release notes](https://github.com/actions/checkout/releases)
- [Changelog](https://github.com/actions/checkout/blob/main/CHANGELOG.md)
- [Commits](https://github.com/actions/checkout/compare/v4...v6)

---
updated-dependencies:
- dependency-name: actions/checkout
  dependency-version: '6'
  dependency-type: direct:production
  update-type: version-update:semver-major
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
Bump actions/setup-python from 5 to 6

Bumps [actions/setup-python](https://github.com/actions/setup-python) from 5 to 6.
- [Release notes](https://github.com/actions/setup-python/releases)
- [Commits](https://github.com/actions/setup-python/compare/v5...v6)

---
updated-dependencies:
- dependency-name: actions/setup-python
  dependency-version: '6'
  dependency-type: direct:production
  update-type: version-update:semver-major
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
Bump globals from 16.5.0 to 17.3.0 in /freedata_gui

Bumps [globals](https://github.com/sindresorhus/globals) from 16.5.0 to 17.3.0.
- [Release notes](https://github.com/sindresorhus/globals/releases)
- [Commits](https://github.com/sindresorhus/globals/compare/v16.5.0...v17.3.0)

---
updated-dependencies:
- dependency-name: globals
  dependency-version: 17.3.0
  dependency-type: direct:development
  update-type: version-update:semver-major
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
Bump i18next-vue from 5.3.0 to 5.4.0 in /freedata_gui

Bumps [i18next-vue](https://github.com/i18next/i18next-vue) from 5.3.0 to 5.4.0.
- [Release notes](https://github.com/i18next/i18next-vue/releases)
- [Commits](https://github.com/i18next/i18next-vue/compare/v5.3.0...v5.4.0)

---
updated-dependencies:
- dependency-name: i18next-vue
  dependency-version: 5.4.0
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
Bump chart.js from 4.5.0 to 4.5.1 in /freedata_gui

Bumps [chart.js](https://github.com/chartjs/Chart.js) from 4.5.0 to 4.5.1.
- [Release notes](https://github.com/chartjs/Chart.js/releases)
- [Commits](https://github.com/chartjs/Chart.js/compare/v4.5.0...v4.5.1)

---
updated-dependencies:
- dependency-name: chart.js
  dependency-version: 4.5.1
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
Bump @babel/eslint-parser from 7.28.5 to 7.28.6 in /freedata_gui

Bumps [@babel/eslint-parser](https://github.com/babel/babel/tree/HEAD/eslint/babel-eslint-parser) from 7.28.5 to 7.28.6.
- [Release notes](https://github.com/babel/babel/releases)
- [Changelog](https://github.com/babel/babel/blob/main/CHANGELOG.md)
- [Commits](https://github.com/babel/babel/commits/v7.28.6/eslint/babel-eslint-parser)

---
updated-dependencies:
- dependency-name: "@babel/eslint-parser"
  dependency-version: 7.28.6
  dependency-type: direct:development
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

Mashintime committed:
Fix a logger error message on win everytime an api call is mode

dependabot[bot] committed:
Bump actions/upload-artifact from 6 to 7

Bumps [actions/upload-artifact](https://github.com/actions/upload-artifact) from 6 to 7.
- [Release notes](https://github.com/actions/upload-artifact/releases)
- [Commits](https://github.com/actions/upload-artifact/compare/v6...v7)

---
updated-dependencies:
- dependency-name: actions/upload-artifact
  dependency-version: '7'
  dependency-type: direct:production
  update-type: version-update:semver-major
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
Bump docker/build-push-action from 6 to 7

Bumps [docker/build-push-action](https://github.com/docker/build-push-action) from 6 to 7.
- [Release notes](https://github.com/docker/build-push-action/releases)
- [Commits](https://github.com/docker/build-push-action/compare/v6...v7)

---
updated-dependencies:
- dependency-name: docker/build-push-action
  dependency-version: '7'
  dependency-type: direct:production
  update-type: version-update:semver-major
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
Bump docker/setup-buildx-action from 3 to 4

Bumps [docker/setup-buildx-action](https://github.com/docker/setup-buildx-action) from 3 to 4.
- [Release notes](https://github.com/docker/setup-buildx-action/releases)
- [Commits](https://github.com/docker/setup-buildx-action/compare/v3...v4)

---
updated-dependencies:
- dependency-name: docker/setup-buildx-action
  dependency-version: '4'
  dependency-type: direct:production
  update-type: version-update:semver-major
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
Update nuitka requirement from <=2.8.9 to <=4.0.7

Updates the requirements on [nuitka](https://github.com/Nuitka/Nuitka) to permit the latest version.
- [Changelog](https://github.com/Nuitka/Nuitka/blob/develop/Changelog.rst)
- [Commits](https://github.com/Nuitka/Nuitka/compare/0.3.11a...4.0.7)

---
updated-dependencies:
- dependency-name: nuitka
  dependency-version: 4.0.7
  dependency-type: direct:production
...

Signed-off-by: dependabot[bot] <support@github.com>

DJ2LS committed:
Merge pull request #1055 from DJ2LS/dependabot/github_actions/develop/actions/checkout-6

Bump actions/checkout from 4 to 6

DJ2LS committed:
Merge pull request #1056 from DJ2LS/dependabot/github_actions/develop/actions/setup-python-6

Bump actions/setup-python from 5 to 6

DJ2LS committed:
Merge pull request #1058 from DJ2LS/dependabot/npm_and_yarn/freedata_gui/develop/globals-17.3.0

Bump globals from 16.5.0 to 17.3.0 in /freedata_gui

DJ2LS committed:
Merge pull request #1060 from DJ2LS/dependabot/npm_and_yarn/freedata_gui/develop/chart.js-4.5.1

Bump chart.js from 4.5.0 to 4.5.1 in /freedata_gui

DJ2LS committed:
Merge pull request #1061 from DJ2LS/dependabot/npm_and_yarn/freedata_gui/develop/babel/eslint-parser-7.28.6

Bump @babel/eslint-parser from 7.28.5 to 7.28.6 in /freedata_gui

DJ2LS committed:
Merge pull request #1076 from DJ2LS/dependabot/pip/develop/nuitka-lte-4.0.7

Update nuitka requirement from <=2.8.9 to <=4.0.7

DJ2LS committed:
Merge pull request #1075 from DJ2LS/dependabot/github_actions/develop/docker/setup-buildx-action-4

Bump docker/setup-buildx-action from 3 to 4

DJ2LS committed:
Merge pull request #1074 from DJ2LS/dependabot/github_actions/develop/docker/build-push-action-7

Bump docker/build-push-action from 6 to 7

DJ2LS committed:
Merge pull request #1067 from DJ2LS/dependabot/github_actions/develop/actions/upload-artifact-7

Bump actions/upload-artifact from 6 to 7

DJ2LS committed:
Merge branch 'develop' into dependabot/npm_and_yarn/freedata_gui/develop/i18next-vue-5.4.0

DJ2LS committed:
Merge branch 'main' into develop

dependabot[bot] committed:
Bump robinraju/release-downloader from 1.12 to 1.13

Bumps [robinraju/release-downloader](https://github.com/robinraju/release-downloader) from 1.12 to 1.13.
- [Release notes](https://github.com/robinraju/release-downloader/releases)
- [Commits](https://github.com/robinraju/release-downloader/compare/v1.12...v1.13)

---
updated-dependencies:
- dependency-name: robinraju/release-downloader
  dependency-version: '1.13'
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
Bump docker/setup-qemu-action from 3 to 4

Bumps [docker/setup-qemu-action](https://github.com/docker/setup-qemu-action) from 3 to 4.
- [Release notes](https://github.com/docker/setup-qemu-action/releases)
- [Commits](https://github.com/docker/setup-qemu-action/compare/v3...v4)

---
updated-dependencies:
- dependency-name: docker/setup-qemu-action
  dependency-version: '4'
  dependency-type: direct:production
  update-type: version-update:semver-major
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
Bump docker/login-action from 3 to 4

Bumps [docker/login-action](https://github.com/docker/login-action) from 3 to 4.
- [Release notes](https://github.com/docker/login-action/releases)
- [Commits](https://github.com/docker/login-action/compare/v3...v4)

---
updated-dependencies:
- dependency-name: docker/login-action
  dependency-version: '4'
  dependency-type: direct:production
  update-type: version-update:semver-major
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
Bump pypa/gh-action-pypi-publish from 1.13.0 to 1.14.0

Bumps [pypa/gh-action-pypi-publish](https://github.com/pypa/gh-action-pypi-publish) from 1.13.0 to 1.14.0.
- [Release notes](https://github.com/pypa/gh-action-pypi-publish/releases)
- [Commits](https://github.com/pypa/gh-action-pypi-publish/compare/v1.13.0...v1.14.0)

---
updated-dependencies:
- dependency-name: pypa/gh-action-pypi-publish
  dependency-version: 1.14.0
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
Bump softprops/action-gh-release from 2 to 3

Bumps [softprops/action-gh-release](https://github.com/softprops/action-gh-release) from 2 to 3.
- [Release notes](https://github.com/softprops/action-gh-release/releases)
- [Changelog](https://github.com/softprops/action-gh-release/blob/master/CHANGELOG.md)
- [Commits](https://github.com/softprops/action-gh-release/compare/v2...v3)

---
updated-dependencies:
- dependency-name: softprops/action-gh-release
  dependency-version: '3'
  dependency-type: direct:production
  update-type: version-update:semver-major
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
Update setuptools requirement from >=77.0.3 to >=82.0.1

Updates the requirements on [setuptools](https://github.com/pypa/setuptools) to permit the latest version.
- [Release notes](https://github.com/pypa/setuptools/releases)
- [Changelog](https://github.com/pypa/setuptools/blob/main/NEWS.rst)
- [Commits](https://github.com/pypa/setuptools/compare/v77.0.3...v82.0.1)

---
updated-dependencies:
- dependency-name: setuptools
  dependency-version: 82.0.1
  dependency-type: direct:development
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
Update nuitka requirement from <=4.0.7 to <=4.0.8

Updates the requirements on [nuitka](https://github.com/Nuitka/Nuitka) to permit the latest version.
- [Changelog](https://github.com/Nuitka/Nuitka/blob/develop/Changelog.rst)
- [Commits](https://github.com/Nuitka/Nuitka/compare/0.3.11a...4.0.8)

---
updated-dependencies:
- dependency-name: nuitka
  dependency-version: 4.0.8
  dependency-type: direct:production
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
Bump bootstrap-vue-next from 0.40.9 to 0.45.0 in /freedata_gui

Bumps [bootstrap-vue-next](https://github.com/bootstrap-vue-next/bootstrap-vue-next/tree/HEAD/packages/bootstrap-vue-next) from 0.40.9 to 0.45.0.
- [Release notes](https://github.com/bootstrap-vue-next/bootstrap-vue-next/releases)
- [Changelog](https://github.com/bootstrap-vue-next/bootstrap-vue-next/blob/main/packages/bootstrap-vue-next/CHANGELOG.md)
- [Commits](https://github.com/bootstrap-vue-next/bootstrap-vue-next/commits/nuxt-v0.45.0/packages/bootstrap-vue-next)

---
updated-dependencies:
- dependency-name: bootstrap-vue-next
  dependency-version: 0.45.0
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
Bump marked from 17.0.1 to 18.0.3 in /freedata_gui

Bumps [marked](https://github.com/markedjs/marked) from 17.0.1 to 18.0.3.
- [Release notes](https://github.com/markedjs/marked/releases)
- [Commits](https://github.com/markedjs/marked/compare/v17.0.1...v18.0.3)

---
updated-dependencies:
- dependency-name: marked
  dependency-version: 18.0.3
  dependency-type: direct:production
  update-type: version-update:semver-major
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
Bump gettext-parser from 8.0.0 to 9.0.2 in /freedata_gui

Bumps [gettext-parser](https://github.com/smhg/gettext-parser) from 8.0.0 to 9.0.2.
- [Release notes](https://github.com/smhg/gettext-parser/releases)
- [Changelog](https://github.com/smhg/gettext-parser/blob/main/CHANGELOG.md)
- [Commits](https://github.com/smhg/gettext-parser/compare/v8.0.0...v9.0.2)

---
updated-dependencies:
- dependency-name: gettext-parser
  dependency-version: 9.0.2
  dependency-type: direct:production
  update-type: version-update:semver-major
...

Signed-off-by: dependabot[bot] <support@github.com>

DJ2LS committed:
Merge pull request #1091 from DJ2LS/dependabot/npm_and_yarn/freedata_gui/develop/gettext-parser-9.0.2

DJ2LS committed:
Merge pull request #1090 from DJ2LS/dependabot/npm_and_yarn/freedata_gui/develop/marked-18.0.3

DJ2LS committed:
Merge pull request #1089 from DJ2LS/dependabot/npm_and_yarn/freedata_gui/develop/bootstrap-vue-next-0.45.0

DJ2LS committed:
Merge pull request #1087 from DJ2LS/dependabot/pip/develop/nuitka-lte-4.0.8

DJ2LS committed:
Merge pull request #1086 from DJ2LS/dependabot/pip/develop/setuptools-gte-82.0.1

DJ2LS committed:
Merge pull request #1084 from DJ2LS/dependabot/github_actions/develop/pypa/gh-action-pypi-publish-1.14.0

DJ2LS committed:
Merge pull request #1083 from DJ2LS/dependabot/github_actions/develop/docker/login-action-4

DJ2LS committed:
Merge pull request #1082 from DJ2LS/dependabot/github_actions/develop/docker/setup-qemu-action-4

DJ2LS committed:
Merge pull request #1081 from DJ2LS/dependabot/github_actions/develop/robinraju/release-downloader-1.13

DJ2LS committed:
Merge pull request #1059 from DJ2LS/dependabot/npm_and_yarn/freedata_gui/develop/i18next-vue-5.4.0

DJ2LS committed:
Merge pull request #1085 from DJ2LS/dependabot/github_actions/develop/softprops/action-gh-release-3
